lib/product-import.coffee

```diff
@@ -719,6 +719,8 @@ class ProductImport
           else
             if _.size(result.body.results) > 1
               @logger.warn "Found more than 1 #{refKey} for #{ref.id}"
+            if _.isEmpty(result.body.results)
```

Comment:
Should not you check if result has body at all? See also line 717?

Reply:
We don't encounter the error you're talking about, that's why I don't want to add it. The error comes from this line:
```
_this._cache[refKey][ref.id] = result.body.results[0];
```
So the body is there and it has `result.body.count != 0`.